src/MoBi.Engine/Sbml/CompartmentImporter.cs

```diff
@@ -116,13 +116,18 @@ private void CreateMoleculeProperties(IContainer container)
       /// <summary>
       ///     Creates a Volume Parameter for a MoBi Container. 
       /// </summary>
-      private IEntity CreateVolumeParameter()
+      private IEntity CreateVolumeParameter(Compartment compartment)
       {
-         IFormula formula = ObjectBaseFactory.Create<ConstantFormula>().WithValue(1);
+         var volume = 1.0;
+         if (compartment.isSetVolume())
+            volume = compartment.getVolume();
+         else if (compartment.isSetSize())
+            volume = compartment.getSize();
+         IFormula formula = ObjectBaseFactory.Create<ConstantFormula>().WithValue(volume);
```

Comment:
@abdelr There is in core a FormulaFactory with a `ConstantFormula` method. This should be used here
Specifically, you will see that you need to set the dimension of the formula as well when creating a formula. At the moment this is missing. Dimension of the formula will be  the one from the containing object

Reply:
Thanks for the catch. I did not know about the FormulaFactory ;)